.github/workflows/canary-weekly.yml

```
@@ -0,0 +1,77 @@
+name: Periodic Canary Build
+
+on:
+  workflow_dispatch:
+  schedule:
+    # * is a special character in YAML so you have to quote this string
+    - cron: "17 14 * * WED"
+
+concurrency:
+  group: ${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  linux:
+    # Allow manually-triggered run on fork or scheduled run on main repository:
+    if: >-
+      github.event_name != 'schedule' || !github.event.repository.fork
```

I'd prefer to encapsulate it like this:

```suggestion
    # - this is a fork and manually-triggered, or
    # - this is the main repo, and
    # - we are on the main (or feature) branch
    if: >-
      (
        github.event.repository.fork
        && github.event_name == 'workflow_dispatch'
      )
      || (
        !github.event.repository.fork
        && (
          github.ref_name == 'main'
          || startsWith(github.ref_name, 'feature/')
        )
      )

```